### ISSUE TITLE
rsconnect::deploySite with relative siteDir errs

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rsconnect#297]
```r
rsconnect::deploySite("rmd-site", ...)
Error in rsconnect::deploySite(siteDir = "rmd-site", server = "localhost",  : 
  rmd-site does not exist
```

This is because we `setwd` before normalizing and validating that `siteDir` exists:
https://github.com/rstudio/rsconnect/blob/5a22a0171264c121bf0409c4b2bae02ae39180ac/R/deploySite.R#L32-L33

https://github.com/rstudio/rsconnect/blob/5a22a0171264c121bf0409c4b2bae02ae39180ac/R/deploySite.R#L46-L51

**Comment by aronatkins** [comment]
Workaround: Use an absolute `siteDir` argument.

**Comment by antass** [comment]
Relative paths work if passed via `here`: `rsconnect::deploySite(here("rmd-site"), ...)` should work.

**Comment by hadley** [comment]
I fixed this in passing in #676
